examples/step-44/step-44.cc

```diff
@@ -2001,16 +1998,18 @@ namespace Step44
       {
         fe_values_ref.reinit(cell);
 
-        PointHistory<dim> *lqph =
-          reinterpret_cast<PointHistory<dim>*>(cell->user_pointer());
-
-        Assert(lqph >= &quadrature_point_history.front(), ExcInternalError());
-        Assert(lqph <= &quadrature_point_history.back(), ExcInternalError());
+        // In contrast to that which was previously called for,
+        // in this instance the quadrature point data is specifically
+        // non-modifiable since we will only be accessing data.
+        const std::vector<std::shared_ptr<const PointHistory<dim> > > lqph =
+            const_cast<const CellDataStorage<typename Triangulation<dim>::cell_iterator,
+                                             PointHistory<dim> >&>(quadrature_point_history).get_data(cell);
```

Comment:
there is a const version of `get()` https://www.dealii.org/developer/doxygen/deal.II/classCellDataStorage.html#a47d9eaadf233b3e97bf072ae6efa40a6
so if you have `const std::vector<std::shared_ptr<const PointHistory<dim> > > = quadrature_point_history).get_data(cell)` then compiler should not call non-const version, which return a vector of non-const pointers `std::vector<std::shared_ptr<const PointHistory<dim> > >`.

What exactly did not work? It did not compile? Perhaps it's easier to chat tomorrow about this...